lesson14/main.js

```diff
@@ -0,0 +1,81 @@
+"use strict";
+const wrapper = document.getElementById("js-wrapper");
+const ul = document.getElementById("js-ul");
+const modalBtn = document.getElementById("js-modalBtn");
+modalBtn.classList.add("btn");
+const modal = document.getElementById("js-modal");
+const openModalBtn = document.getElementById("js-openModalBtn");
+const jsonURL = "https://myjson.dit.upm.es/api/bins/j16z";
+const removeModal = () => {
+   modal.remove();
+   wrapper.classList.remove("bg_gray");
+};
+
+const startLoading = () => {
+   const loadingIcon = document.createElement("img");
+   loadingIcon.id = "loadingIcon";
+   loadingIcon.src = "img/loading-circle.gif";
+   wrapper.appendChild(loadingIcon);
+};
+
+const endLoading = () => {
+   const loadingIcon = document.getElementById("loadingIcon");
+   loadingIcon.remove();
+}
+
+openModalBtn.addEventListener("click", function () {
+   modal.style.display = "block";
+   wrapper.classList.add("bg_gray");
+   openModalBtn.remove();
+});
+
+modalBtn.addEventListener("click", () => {
+   const inputNumber = document.getElementById("js-number");
+   if (inputNumber.value === "") {
+      alert("数字が未入力です。");
+      return;
+   }
+   console.log(inputNumber.value);
+   init();
+});
+
+async function init() {
+   removeModal();
+   startLoading();
+   try {
+      const result = await fetchData();
+      createListElements(result);
+   } catch (e) {
+      const div = document.createElement("div");
+      ul.appendChild(div);
+      div.textContent = ` エラーが発生しました：${e}`;
+      console.log(e);
+   } finally {
+      endLoading();
+   }
```

Comment:
IMO
Sorry, I forgot to share what Morita-san taught me.
And add what happens when the data we get is empty! This will make the code more friendly.

```suggestion
   try {
      const result = await fetchData();
      if (result.length === 0) {
      throw new Error("データが空です");
    }
      createListElements(result);
   } catch (e) {
      const div = document.createElement("div");
      ul.appendChild(div);
      div.textContent = ` エラーが発生しました：${e}`;
      console.log(e);
   } finally {
      endLoading();
   }
```
[データが空だったときのsandbox](https://codesandbox.io/s/gifted-hugle-fl23p?file=/main.js)

Reply:
Thank you for your repeated reviews.
I didn't thought that far ahead. 
I fixed it.
f44720e